lib/target.mli

```diff
@@ -2,23 +2,44 @@ open Value
 open Env
 open Types
 
-type extern = EvalEnv.t -> value list -> EvalEnv.t * value 
+type 'st assign = 
+  ctrl -> EvalEnv.t -> 'st -> lvalue -> value -> EvalEnv.t * 'st * signal
+
+type ('st1, 'st2) pre_extern =
+  'st1 assign -> ctrl -> EvalEnv.t -> 'st2 -> Type.t list -> value list ->
+  EvalEnv.t * 'st2 * signal * value
+
+module State : sig
+  type 'a t
+
+  val empty : 'a t
+  val insert : int -> 'a -> 'a t -> 'a t
+  val find : int -> 'a t -> 'a
+  val fresh_loc : unit -> int

```

Comment:
We should find a way to statically allocate programs to avoid having to call fresh_loc like we talked about. But that's another PR's worth of hacking I think

Reply:
yeah this cant happen until i refactor on the new AST types